sim/dex-conditions.ts

```diff
@@ -1,21 +1,52 @@
 import {BasicEffect, toID} from './dex-data';
-import type {SecondaryEffect, MoveEventMethods} from './dex-moves';
+import type {SecondaryEffect, MoveSpecificEventMethods} from './dex-moves';
 
+type GenericEvent = 'DamagingHit' | 'EmergencyExit' | 'AfterEachBoost' | 'AfterHit' | 'AfterMega' | 'AfterSetStatus' |
+'AfterSubDamage' | 'AfterSwitchInSelf' | 'AfterUseItem' | 'AfterBoost' | 'AfterFaint' | 'AfterMoveSecondarySelf' |
+'AfterMoveSecondary' | 'AfterMove' | 'AfterMoveSelf' | 'Attract' | 'Accuracy' | 'BasePower' | 'BeforeFaint' |
+'BeforeMove' | 'BeforeSwitchIn' | 'BeforeSwitchOut' | 'BeforeTurn' | 'Boost' | 'ChargeMove' | 'CriticalHit' |
+'Damage' | 'DeductPP' | 'DisableMove' | 'DragOut' | 'EatItem' | 'Effectiveness' | 'EntryHazard' | 'Faint' | 'Flinch' |
+'FractionalPriority' | 'Hit' | 'Immunity' | 'LockMove' | 'MaybeTrapPokemon' | 'ModifyAccuracy' | 'ModifyAtk' |
+'ModifyBoost' | 'ModifyCritRatio' | 'ModifyDamage' | 'ModifyDef' | 'ModifyMove' | 'ModifyPriority' |
+'ModifySecondaries' | 'ModifyType' | 'ModifyTarget' | 'ModifySpA' | 'ModifySpD' | 'ModifySpe' | 'ModifyWeight' |
+'MoveAborted' | 'NegateImmunity' | 'OverrideAction' | 'PrepareHit' | 'RedirectTarget' | 'Residual' | 'SetAbility' |
+'SetStatus' | 'SetWeather' | 'StallMove' | 'SwitchIn' | 'SwitchOut' | 'Swap' | 'TakeItem' | 'WeatherStart' |
+'TrapPokemon' | 'TryAddVolatile' | 'TryEatItem' | 'TryHeal' | 'TryHit' | 'TryHitField' | 'TryHitSide' |
+'Invulnerability' | 'TryMove' | 'TryPrimaryHit' | 'Type' | 'Update' | 'Weather' | 'WeatherModifyDamage' |
+'ModifyDamagePhase1' | 'ModifyDamagePhase2';
```

Comment:
Would it be worthwhile to furher section this? i.e. making  subtype called `ModifierEvents` and having `ModifyAtk` etc on them, or no?

Reply:
I commented with the intent that it would mostly just be for organization but it’s fine either way so I won’t push